examples/dwarfdump.rs

```diff
@@ -65,6 +71,57 @@ impl From<io::Error> for Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
+fn parallel_output<II: IntoIterator, F>(max_workers: usize, iter: II, f: F) -> Result<()>
+    where F: Sync + Fn(II::Item, &mut Vec<u8>) -> Result<()>,
+          II::IntoIter: Send {
+    let state = Mutex::new((iter.into_iter().fuse(), 0, Ok(())));
```

Comment:
Having the `fuse` doesn't hurt, but is it actually needed for the current caller?

Reply:
I think this code is complex enough that using a local struct type for `state` would improve readability.

Reply:
We depend on fused behavior. If we remove `fuse` here then I would want to make parallel_output take a `Vec<T>` instead of a generic `Iterator`/`IntoIterator`. Is that preferable?

Reply:
No, it's fine how it is.